### ISSUE TITLE
Is valid_set needed？

### ISSUE THREAD

**Issue by bayou3** [thunlp/OpenMatch#1]
Hi, I saw there is the train_set and dev_set in the "train.sh", dev_set is the test_set. So is valid_set necessary?

**Comment by zkt12** [comment]
Yes, but you can remove it in train.py if you don't need.

**Comment by bayou3** [comment]
Thanks.
For using valide_set, I think some lines like below can be added in the function of train_reinfoselect( ) of train.py, right?
`best_mes = 0.0
 with torch.no_grad(): 
    valid_mes = dev(args, model, metric, valid_loader, device)
    if valid_mes[args.metric] > best_mes:
       dev_mes = dev(args, model, metric, dev_loader, device)
       best_mes = dev_mes[args.metric]
`

**Comment by zkt12** [comment]
Yes, for reinfoselect training, the valid_set is used as target dataset, the gains of the valid_mes is the reward for selection policy optimizing.

**Comment by bayou3** [comment]
Thank you!

**Comment by zkt12** [comment]
dev is valid_set, test_set is used in inference :)

**Comment by bayou3** [comment]
Yep！Thank you for pointing out and correcting me!
